### ISSUE TITLE
Grooyscript error running under Cleanroom loader

### ISSUE THREAD

**Issue by lynxx131** [Nomi-CEu/Nomi-CEu#841]
### Nomi CEu Version

1.7-beta-5c

### Launcher

Prism Launcher

### Configurations or Mods Changed

Running Cleanroom build https://github.com/CleanroomMC/Cleanroom/actions/runs/10138104505 with latest Fugue and Scalar 2.11.

Client is running on Windows with Java 21.

### Environment

Singleplayer

### Mode

Expert Mode

### What Happened

Receive a GroovyScript error upon entering the world:

```
============  GroovyLog  ====  31.07.2024  ============
GroovyScript version: 1.1.1
[01:12:15] [CLIENT/INFO] [nomiceu]: Running scripts in loader 'preInit'
[01:13:11] [CLIENT/INFO] [nomiceu]: Running scripts in loader 'init'
[01:13:13] [CLIENT/INFO] [nomiceu]: Running scripts in loader 'postInit'
[01:13:13] [CLIENT/INFO] [nomiceu]:  - running org.codehaus.groovy.runtime.InvokerHelper$2
[01:13:13] [CLIENT/INFO] [nomiceu]:  - running earlygame
[01:13:13] [CLIENT/ERROR] [nomiceu]: An exception occurred while running scripts. Look at latest.log for a full stacktrace:
	java.lang.NoClassDefFoundError: net/minecraft/item/ItemStack
		at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
		at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3578)
		at java.base/java.lang.Class.privateGetPublicMethods(Class.java:3603)
		at java.base/java.lang.Class.getMethods(Class.java:2185)
		at java.desktop/com.sun.beans.introspect.MethodInfo.get(MethodInfo.java:89)
		at java.desktop/com.sun.beans.introspect.ClassInfo.getMethods(ClassInfo.java:82)
		at java.desktop/java.beans.Introspector.getTargetMethodInfo(Introspector.java:1029)
		at java.desktop/java.beans.Introspector.getBeanInfo(Introspector.java:446)
		at java.desktop/java.beans.Introspector.getBeanInfo(Introspector.java:195)
		at groovy.lang.MetaClassImpl.lambda$addProperties$25(MetaClassImpl.java:3466)
		at java.base/java.security.AccessController.doPrivileged(AccessController.java:571)
		at groovy.lang.MetaClassImpl.doPrivileged(MetaClassImpl.java:3519)
		at groovy.lang.MetaClassImpl.addProperties(MetaClassImpl.java:3466)
		at groovy.lang.MetaClassImpl.reinitialize(MetaClassImpl.java:3448)
		at groovy.lang.MetaClassImpl.initialize(MetaClassImpl.java:3441)
		at org.codehaus.groovy.reflection.ClassInfo.getMetaClassUnderLock(ClassInfo.java:272)
		at org.codehaus.groovy.reflection.ClassInfo.getMetaClass(ClassInfo.java:314)
		in earlygame.groovy in line 20
		at com.cleanroommc.groovyscript.sandbox.GroovySandbox.runScript(GroovySandbox.java:166)
		at com.cleanroommc.groovyscript.sandbox.GroovyScriptSandbox.runScript(GroovyScriptSandbox.java:169)
		at com.cleanroommc.groovyscript.sandbox.GroovySandbox.loadScripts(GroovySandbox.java:139)
```

I attempted to update GroovyScript to 1.1.2 and NomiLabs neither of which worked.

Please let me know if I can provide any further info or testing.

### Expected Behavior

No GrooyScript error on login

### Reproduction Steps

1. Launch Nomifactory with linked Cleanroom build with Scalar and Fugue
2. Create new world
3. Observe GroovyScript error in chat

### Logs

[groovy.log](https://github.com/user-attachments/files/16430263/groovy.log)
[latest.log](https://github.com/user-attachments/files/16430264/latest.log)

**Comment by lynxx131** [comment]
Just an update I think this might only be present in Singleplayer as I don't get any errors in the groovy log when running on a server.

**Comment by IntegerLimit** [comment]
I think this might be GroovyScript failing to obfuscate the scripts for Cleanroom. Will investigate.

**Comment by lynxx131** [comment]
Cool, also ignore my above comment turns out it does do it on the server side as well I was just blind :).

**Comment by HandsomeAsianMan** [comment]
downgrade groovy to 1.1.1

**Comment by IntegerLimit** [comment]
GrS 1.1.1 should work fine, without errors, but it will have problems with `/gs reload` on any version above 1.7-beta-5c.

**Comment by kaspertfg** [comment]
I have also tried this with the latest nightly and get the same error as well.  
Groovy log
https://mclo.gs/0MiQETk
Latest log
https://mclo.gs/L8p4soY

But considering that there is also betterquesting/cleanroom bug that causes crashes with the EBF quest as well, I wouldn't suggest using cleanroom till that is also fixed.

**Comment by HandsomeAsianMan** [comment]
This is such an odd bug cuz it works perfectly fine for me.

**Comment by kaspertfg** [comment]
> This is such an odd bug cuz it works perfectly fine for me.

What version of cleanroom, fugue, scala, and java are you using?

**Comment by HandsomeAsianMan** [comment]
<@554533282986065921> I cant seem to get the error???
1. installed nomi 1.7.5c
2. unzip and moved [patches, libraries, mmc-pack.json] into nomi 1.7.5c instance
3. installed scalar 2.11.1 and fugue 0.15.9
4. disabled MixinBooter and ConfigAnytime

Java is Adoptium 21
Clean room version is latest release https://github.com/CleanroomMC/Cleanroom/releases/download/0.2.2-alpha/Cleanroom-MMC-instance-0.2.2-alpha.zip

**Comment by HandsomeAsianMan** [comment]
Chat error only comes up when updating groovy to 1.1.2

**Comment by lynxx131** [comment]
So I've just tested with an instance of 1.7-beta5c downloaded from Curse via Prism Launcher with the CleanroomMMC 0.2.2 Alpha zip and Fugue 0.15.9 and Scalar 2.11 as per @HandsomeAsianMan's comment and here's what I've found.

1. First launch works perfectly fine, no chat error on world join
2. Close and re-launch the game and join a world, you get the chat error
3. Close the game, delete the `cache` folder inside `.minecraft` and re-launch, no chat error

So looks like something around the cache files that GroovyScript is generating because clearing them removes the chat error.
